generate-release-notes.py

```diff
@@ -1,20 +1,39 @@
 #!/usr/bin/env python3
-from github import Github  # https://github.com/PyGithub/PyGithub
-import yaml
+# pylint: disable=missing-module-docstring
+# pylint: disable=missing-class-docstring
+# pylint: disable=missing-function-docstring
 import os
 import re
 import io
 import datetime
 
+# To install, run python3 -m pip install PyYAML
+import yaml
+
+# To install, run python3 -m pip install PyGithub==2.3.0
+from github import Github, Auth  # https://github.com/PyGithub/PyGithub
+
+# Validate required environment variables
+GITHUB_TOKEN = os.environ.get("GITHUB_TOKEN")
+assert GITHUB_TOKEN, "GITHUB_TOKEN must be set"
+VERSION = os.environ.get("VERSION")
+assert os.environ.get("VERSION"), "VERSION must be set"
+assert VERSION.startswith("v"), "VERSION must start with 'v'"
+
 # First create a Github instance. Create a token through Github website - provide "repo" auth.
-g = Github(os.environ.get('GITHUB_TOKEN'))
+auth = Auth.Token(os.environ.get("GITHUB_TOKEN"))
+g = Github(auth=auth)
 
 # The milestone to generate notes for.
-assert os.environ.get('VERSION')
-MILESTONE=os.environ.get('VERSION')
+MILESTONE = VERSION
```

Comment:
If `MILESTONE` is always going to == `VERSION` we should replace `MILESTONE` with `VERSION` for clarity.